File: rb/features.rb
# FussyApiDocumentation SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module FussyApiDocumentationFeatures
  def self.make_feature(name)
    case name
    when "base"
      FussyApiDocumentationBaseFeature.new
    when "ratelimit"
      FussyApiDocumentationRatelimitFeature.new
    when "retry"
      FussyApiDocumentationRetryFeature.new
    when "test"
      FussyApiDocumentationTestFeature.new
    when "timeout"
      FussyApiDocumentationTimeoutFeature.new
    else
      FussyApiDocumentationBaseFeature.new
    end
  end
end

File: py/features.py
# FussyApiDocumentation SDK feature factory

from feature.base_feature import FussyApiDocumentationBaseFeature
from feature.test_feature import FussyApiDocumentationTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FussyApiDocumentationBaseFeature(),
        "test": lambda: FussyApiDocumentationTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
